examples/oemof_0.2/storage_investment/storage_investment_1.py

```diff
@@ -0,0 +1,178 @@
+excess = solph.Sink(label='excess_bel', inputs={bel: solph.Flow()})
+
+# create source object representing the natural gas commodity (annual limit)
+gas_resource = solph.Source(label='rgas', outputs={bgas: solph.Flow(
+    variable_costs=price_gas)})
+
+# create fixed source object representing wind power plants
+wind = solph.Source(label='wind', outputs={bel: solph.Flow(
+    actual_value=data['wind'], fixed=True,
+    investment=solph.Investment(ep_costs=epc_wind))})
+
+# create fixed source object representing pv power plants
+pv = solph.Source(label='pv', outputs={bel: solph.Flow(
+    actual_value=data['pv'], fixed=True,
+    investment=solph.Investment(ep_costs=epc_pv))})
+
+# create simple sink object representing the electrical demand
+demand = solph.Sink(label='demand', inputs={bel: solph.Flow(
+    actual_value=data['demand_el'], fixed=True, nominal_value=1)})
+
+# create simple transformer object representing a gas power plant
+pp_gas = solph.Transformer(
+    label="pp_gas",
+    inputs={bgas: solph.Flow()},
+    outputs={bel: solph.Flow(nominal_value=10e10, variable_costs=0)},
+    conversion_factors={bel: 0.58})
+
+# my_list = [0.1] * number_timesteps
+
+# create storage object representing a battery
+storage = solph.components.GenericStorage(
+    label='storage',
+    # min=my_list,
+    inputs={bel: solph.Flow(variable_costs=0.0001)},
+    outputs={bel: solph.Flow()},
+    capacity_loss=0.00, initial_capacity=0,
+    nominal_input_capacity_ratio=1/6,
+    nominal_output_capacity_ratio=1/6,
+    inflow_conversion_factor=1, outflow_conversion_factor=0.8,
+    investment=solph.Investment(ep_costs=epc_storage),
+)
+
+energysystem.add(excess, gas_resource, wind, pv, demand, pp_gas, storage)
+
+##########################################################################
+# Optimise the energy system
+##########################################################################
+
+logging.info('Optimise the energy system')
+
+# initialise the operational model
+om = solph.Model(energysystem)
+
+# if tee_switch is true solver messages will be displayed
+logging.info('Solve the optimization problem')
+om.solve(solver='cbc', solve_kwargs={'tee': True})
+
+##########################################################################
+# Check and plot the results
+##########################################################################
+
+# check if the new result object is working for custom components
+results = processing.results(om)
+
+custom_storage = views.node(results, 'storage')
+electricity_bus = views.node(results, 'electricity')
+
+meta_results = processing.meta_results(om)
+pp.pprint(meta_results)
+
+my_results = electricity_bus['scalars']
+my_results['storage_invest_GWh'] = results[(storage, None)]['scalars']['invest']/1e6
+my_results['wind_invest_MW'] = results[(wind, bel)]['scalars']['invest']/1e3
+my_results['res_share'] = 1 - results[(pp_gas, bel)]['sequences'].sum()/results[(bel, demand)]['sequences'].sum()
```

Comment:
This returns a dataframe where I would only select the flow.

Reply:
I added some explanation.

>my_results['res_share']['flow']

This one I'm going to postpone to #13